datafusion/physical-expr/src/expressions/in_list.rs

```diff
@@ -398,6 +357,26 @@ fn set_contains_utf8<OffsetSize: OffsetSizeTrait>(
     collection_contains_check!(array, native_set, negated, contains_null)
 }
 
+fn set_contains_binary<OffsetSize: OffsetSizeTrait>(
+    array: &GenericBinaryArray<OffsetSize>,
+    set: &HashSet<ScalarValue>,
+    negated: bool,
+) -> ColumnarValue {
+    let contains_null = set.iter().any(|v| v.is_null());
+    let native_array = set
+        .iter()
+        .flat_map(|v| match v {
+            Binary(v) | LargeBinary(v) => v.as_deref(),
+            datatype => {
+                unreachable!("InList can't reach other data type {} for {}.", datatype, v)
+            }
+        })
+        .collect::<Vec<_>>();
+    let native_set: HashSet<&[u8]> = HashSet::from_iter(native_array);
```

Comment:
Can you skip this intermediate `Vec` and instead go directly to HashSet?

Like `.collect::<HashSet<_>>()` ?

Reply:
I will do this in a follow-up patch as we have this problem in both `binary` and `string`